target/linux/ramips/dts/mt7621_iodata_wnpr2600g.dts

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "iodata,wnpr2600g", "mediatek,mt7621-soc";
+	model = "I-O DATA WNPR2600G";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "wnpr2600g:green:power";
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		notification {
+			label = "wnpr2600g:green:notification";
+			gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan2g {
+			label = "wnpr2600g:green:wlan2g";
+			gpios = <&gpio0 14 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		wlan5g {
+			label = "wnpr2600g:green:wlan5g";
+			gpios = <&gpio0 15 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy1radio";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		auto {
+			label = "auto";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x030000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x030000 0x010000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x040000 0x010000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x050000 0xda0000>;
+			};
+
+			partition@df0000 {
+				label = "manufacture";
+				reg = <0xdf0000 0x190000>;
+				read-only;
+			};
+
+			partition@f80000 {
+				label = "storage";
+				reg = <0xf80000 0x080000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x4>;
+	mediatek,portmap = "wllll";
```

Comment:
I've had a look at the whole "mediatek,portmap" story some time ago. If I'm not mistaken, this will not work.

If you look at https://github.com/openwrt/openwrt/blob/34e22653ac18b6ac7fd368ca47625f665808067f/target/linux/ramips/files-4.14/drivers/net/ethernet/mediatek/esw_rt3050.c#L1366 , you will see that only 32 bit integers are valid here. It just "fails" silently.

So TL;DR: 
Use <0x3e> to fit the current scheme.

This value is however only "needed" for displaying a "AN flag in swconfig. It affects no functionality / configuration. Also - the mapping is a bit misleading, as currently a lot of devices have this property either declared incorrect or even incorrect (as a String).

To add insult to injury, the LAN flag is very often (so in my suggestion) set for the CPU port, which is strange as this port carries LAN and WAN separated by a VLAN tag.

So we should maybe consider dropping this property globally altogether, as we have a cleaner track in the userspace (and the property isn't used upstream).

Reply:
Thank you for your review. However, I recognize that ```of_property_read_string()``` is used in MT7530 switch, but is it wrong?

https://github.com/openwrt/openwrt/blob/920abb9b5865fe0994a5c8115cde616dac63d5ae/target/linux/ramips/files-4.14/drivers/net/ethernet/mediatek/mt7530.c#L238

Reply:
Okay, i was a bit blind there :) You are right, for the mt7620 it's a string then. Hooray for consistency.

However, if I'm not mistaken, the actual config is overwritten afterwards by userspace. In addition, only the predefined portmaps seem to be supported (But I'm not sure how much of an issue this is).